### PR TITLE
[faucet] Timeout waiting requests more aggressively

### PR DESCRIPTION
**Pull request by amnn**
- Move the timeout middleware higher up in the stack so requests that are stuck waiting for a long time get aborted instead.
- Shorten the default timeout from 120s to 30s.
- Consolidate the "load shedding" and "timeout" error responses, because we don't use load shedding, and the timeout is probably a signal that the service is overloaded and the request should be issued again, later.

## Test Plan

```
~/sui/crates/sui-faucet$ cargo nextest run
```

### CODE REVIEW

File: crates/sui-faucet/src/main.rs [hunk 1]
```diff
@@ -50,7 +50,7 @@ struct FaucetConfig {
     #[clap(long, default_value_t = 10)]
     request_buffer_size: usize,
 
-    #[clap(long, default_value_t = 120)]
+    #[clap(long, default_value_t = 30)]
```
Comment (longbowlu) [inline crates/sui-faucet/src/main.rs:53]: 30s is probably still too long. Normal users (who request coins from wallet UI) wouldn't wait for 30 seconds before getting frastrated

**Comment by longbowlu** [comment]
A simple way to test it: add some sleep code in faucet, maybe like 60 seconds, and fire >= 5 and >= 15 (5+10 buffer) and see whether it times out correctly

**Review by longbowlu — APPROVED** [review]

File: crates/sui-faucet/src/main.rs (hunk 1 above)
Comment (amnn) [reply crates/sui-faucet/src/main.rs:53]: My thinking here is that timeout should be a last line of defence, and we should have alerting in place at, e.g. the 10s mark so we know when service has degraded to a point that is irritating for people using wallet.  Then having a window between 10s and 30s gives us a chance to weather hot spots in our load without doing anything (apart from knowing about it).

i.e. I don't want to set the timeout to the maximum acceptable time (which I think falls in the 5-10s range).  Thoughts?

**Comment by amnn** [comment]
Timeout doesn't do what we want unfortunately: If a request hits the timeout, it fires off a timeout response to the client, but the faucet is still doing the long-running work (serving the request, or letting it sit in a queue).  So we can end up in situations where the request times out, but the faucet later issues gas to the requested recipient (lose-lose).

Will submit a new PR, replacing it with load-shedding.

**Pull request closed without merging**